test/test_automl/test_automl.py

```diff
@@ -933,3 +935,184 @@ def predict(self, x):
         _model_predict(model, X, task, logger=None)
 
         assert len(w) == 1, "One warning sould have been emmited"
+
+
+@pytest.mark.parametrize("dataset_compression", [False])
+def test_param_dataset_compression_false(dataset_compression: bool) -> None:
+    """
+    Parameters
+    ----------
+    dataset_compression: bool
+        The dataset_compression arg set as False
+
+    Expects
+    -------
+    * Should set the private attribute to None
+    """
+    auto = AutoMLRegressor(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        dataset_compression=dataset_compression
+    )
+
+    assert auto._dataset_compression is None
+
+
+@pytest.mark.parametrize("dataset_compression", [True])
+def test_construction_param_dataset_compression_true(dataset_compression: bool) -> None:
+    """
+    Parameters
+    ----------
+    dataset_compression: bool
+        The dataset_compression arg set as True
+
+    Expects
+    -------
+    * Should set the private attribute to the defaults
+    """
+    auto = AutoMLRegressor(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        dataset_compression=dataset_compression
+    )
+
+    assert auto._dataset_compression == default_dataset_compression_arg
+
+
+@pytest.mark.parametrize("dataset_compression", [{"memory_allocation": 0.2}])
+def test_construction_param_dataset_compression_valid_dict(dataset_compression: Dict) -> None:
+    """
+    Parameters
+    ----------
+    dataset_compression: Dict
+        The dataset_compression arg set partially specified
+
+    Expects
+    -------
+    * Should set the private attribute to the passed dataset_compression arg + defaults
+    """
+    auto = AutoMLRegressor(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        dataset_compression=dataset_compression
+    )
+
+    expected_memory_allocation = dataset_compression["memory_allocation"]
+    expected_methods = default_dataset_compression_arg["methods"]
+
+    assert auto._dataset_compression is not None
+    assert auto._dataset_compression["memory_allocation"] == expected_memory_allocation
+    assert auto._dataset_compression["methods"] == expected_methods
+
+
+@pytest.mark.parametrize("dataset_compression", [{"methods": ["precision", "subsample"]}])
+@pytest.mark.parametrize("X", [np.ones((100, 10), dtype=int)])
+@pytest.mark.parametrize("y", [np.random.random((100,))])
+@unittest.mock.patch("autosklearn.automl.reduce_dataset_size_if_too_large")
+def test_fit_performs_dataset_compression_without_precision_with_int(
```

Comment:
Could you please explain the motivation why we don't reduce the precision in case of integers?

Reply:
Suppose someone provides an int `int64.max`, there is no meaningful way to reduce precision to 32 bits without losing a lot of information. 

In fact, it would turn the the integer negative in `int32`.

Reply:
I realise you probably mean to describe this in the test, I'll add it there

Reply:
Maybe we should then change ints to floats and then reduce the precision? But let's not go to deep into this as I don't think there's currently any need to further look into this.